examples/ttf.zig

```diff
@@ -0,0 +1,171 @@
+            metrics.minx, metrics.maxx, metrics.miny, metrics.maxy, metrics.advance,
+        });
+    } else |err| {
+        std.log.warn("Could not get glyph metrics for 'A': {s}", .{@errorName(err)});
+    }
+    if (font.getGlyphKerning('V', 'A')) |kerning| {
+        std.log.info("Kerning for 'VA': {d}", .{kerning});
+    } else |err| {
+        std.log.warn("Could not get glyph kerning for 'VA': {s}", .{@errorName(err)});
+    }
+
+    const white: sdl3.ttf.Color = .{ .r = 255, .g = 255, .b = 255, .a = 255 };
+    const yellow: sdl3.ttf.Color = .{ .r = 255, .g = 255, .b = 0, .a = 255 };
+    const cyan: sdl3.ttf.Color = .{ .r = 0, .g = 255, .b = 255, .a = 255 };
+    const magenta: sdl3.ttf.Color = .{ .r = 255, .g = 0, .b = 255, .a = 255 };
+
+    const solid_texture = try textureFromSurface(renderer, try font.renderTextSolid("Solid Text", white));
+    defer solid_texture.deinit();
+
+    const shaded_texture = try textureFromSurface(renderer, try font.renderTextShaded("Shaded Text", yellow, .{ .r = 50, .g = 50, .b = 50, .a = 255 }));
+    defer shaded_texture.deinit();
+
+    const blended_texture = try textureFromSurface(renderer, try font.renderTextBlended("Blended Text", cyan));
+    defer blended_texture.deinit();
+
+    const wrapped_text = "This is a looooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooong text that will be wrapped.";
+    const wrapped_texture = try textureFromSurface(renderer, try font.renderTextBlendedWrapped(wrapped_text, magenta, 400));
+    defer wrapped_texture.deinit();
+
+    font.setStyle(.{ .bold = true, .italic = true });
+    const styled_texture = try textureFromSurface(renderer, try font.renderTextBlended("Bold and Italic", white));
+    defer styled_texture.deinit();
+    font.setStyle(.{}); // Reset.
+
+    try font.setOutline(2);
+    const outlined_texture = try textureFromSurface(renderer, try font.renderTextBlended("Outlined Text", yellow));
+    defer outlined_texture.deinit();
+    try font.setOutline(0); // Reset.
+
+    const long_text = "This is a very long string that we want to fit into a small space.";
+    const max_width = 200;
+    const measured = try font.measureString(long_text, max_width);
+    const truncated_text = try std.fmt.allocPrint(allocator, "{s}...", .{long_text[0..measured.measured_length]});
+    defer allocator.free(truncated_text);
+    const truncated_texture = try textureFromSurface(renderer, try font.renderTextBlended(truncated_text, white));
+    defer truncated_texture.deinit();
+
+    const text_engine: sdl3.ttf.RendererTextEngine = try .init(renderer);
+    defer text_engine.deinit();
+
+    const text_obj: sdl3.ttf.Text = try .init(.{ .value = text_engine.value }, font, "Editable Text Object");
+    defer text_obj.deinit();
+    try text_obj.setColor(255, 165, 0, 255);
+    try text_obj.setPosition(10, 450);
+
+    var fps_capper = sdl3.extras.FramerateCapper(f32){ .mode = .{ .limited = 60 } };
```

Comment:
It'd be best to use vsync on the renderer first and have this be unlimited. Only if setting vsync fails should we use the fixed FPS.